veggieseasons/lib/screens/search.dart

```diff
@@ -61,12 +61,18 @@ class _SearchScreenState extends State<SearchScreen> {
     }
 
     return ListView.builder(
-      itemCount: veggies.length,
+      itemCount: veggies.length + 1,
       itemBuilder: (context, i) {
-        return Padding(
-          padding: EdgeInsets.only(left: 16, right: 16, bottom: 24),
-          child: VeggieHeadline(veggies[i]),
-        );
+        if (i == 0) {
```

Comment:
I'm a bit concerned about hardcoding the height (64) the search bar occupies here. Is it possible to use a [SliverPersistentHeader](https://master-api.flutter.dev/flutter/widgets/SliverPersistentHeaderDelegate-class.html) to achieve this?

Reply:
Doesn't seem easy to do with `SliverPersistentHeader` since it currently requires us to specify `minExtent` and `maxExtent`. /cc @Piinks

Reply:
Respected sir,
I am just a budding developer and am glad that I can ask for your guidance at this step. Sir my thinking was since other things used to create the search bar like paddings,margins have been hard coded and default size of icon which remains same across platforms has been used wouldn't the size of searchbar automatically get fixed and if the height is fixed then hard coding this value should not be a problem as if it works fine in one device it will work similarly in others as well.

Reply:
hi kjain333, indeed the `SearchBar` has a fixed intrinsic height, but the height value may change over time. Other contributors may adjust the paddings/margins of the `SearchBar` widget, without realizing that could break the UI of the search page. IMO we should promote a more robust approach.

There doesn't seem to be a readily available widget that does exactly what we want, unfortunately. The use case here doesn't seem to be niche so maybe we should introduce a new `Sliver` widget in flutter?

Reply:
Respected Sir,
I completely agree with the comment you have made and accept my mistake for not actually thinking about the same. It would be really nice if we would create a new Sliver widget for the same and hopefully I would be able to contribute to that as well. Till then I will also try to think of a work around which can hopefully solve this issue.

Reply:
Respected sir,
Sorry to bother you again. I understand that you are trying to build a robust approach towards solving this problem but using the tools we presently have will it be a good approach to replace the SizedBox with the SearchBox itself but with opacity 0 so that it occupies the same space and wouldn't affect the design as well?

Reply:
Could we just remove the SizedBox? It does not appear to be necessary.

If we wrap the _createSearchBox in a`Visibility` widget, that will achieve the invisible placeholder, with the added benefit that we know it won't be interacted with.

```dart
return ListView.builder(
      itemCount: veggies.length,
      itemBuilder: (context, i) {
        if (i == 0) {
          return Visibility(
            child: _createSearchBox(),
            visible: false,
            maintainSize: true,
            maintainAnimation: true,
            maintainState: true,
          );
        }
        return Padding(
          padding: EdgeInsets.only(left: 16, right: 16, bottom: 24),
          child: VeggieHeadline(veggies[i]),
        );
      },
    );
```

I've run this, and it appears to WAI. :)

Reply:
Respected Ma'am,
It's really good to know that we are on the same page with the change of things. With the latest change I have made to my request I have used the Opacity widget which indeed makes the search bar invisible but the part I missed was that it should not even be click able even though very less people would be able to get it that there is a search bar there as well. 
Thanks a lot for pointing this out and I will make these changes right away.

Reply:
That's clever! I'm on board with this approach. But still I think it's a little bit hacky, and this is a common enough use case that `SliverPersistentHeader` should probably support. Maybe we can retrofit this when we have the right sliver.

Reply:
True! I don't know that we are missing a sliver here. It can definitely be done with a SliverPersistentHeader, a SliverAppBar most likely, it would just mean that we would need to convert the ListView to a SliverList with a CustomScrollView, etc. All that seems a bit more than necessary.

Reply:
It seems like this can't be solved elegantly without creating some kind of render object or other class that would sit above both the ListView and the search box, know how big the latter is, and pass that information down to whatever's building the ListView.

Therefore it's cool by me to merge this as-is, though I'd recommend adding a one line comment above line 68 that explains why the code is building a second, invisible search box.